server/routes/journeys/record-investigation/e2e.cy.ts

```diff
@@ -0,0 +1,268 @@
+    govukTaskListStatusShouldBe(sections.WHY_BEHAVIOUR, 'Completed')
+    cy.findAllByText('Incomplete').should('have.length', 4)
+
+    completeTextboxSection(sections.EVIDENCE, /evidence-secured$/, 'What evidence has been secured?')
+
+    cy.findAllByText('Cannot save yet').should('have.length', 1)
+    govukTaskListStatusShouldBe(sections.DETAILS, 'Completed')
+    govukTaskListStatusShouldBe(sections.STAFF, 'Completed')
+    govukTaskListStatusShouldBe(sections.WHY_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.EVIDENCE, 'Completed')
+    cy.findAllByText('Incomplete').should('have.length', 3)
+
+    completeTextboxSection(
+      sections.USUAL_BEHAVIOUR,
+      /usual-behaviour-presentation$/,
+      'What is Testname User’s usual behaviour presentation?',
+    )
+
+    cy.findAllByText('Cannot save yet').should('have.length', 1)
+    govukTaskListStatusShouldBe(sections.DETAILS, 'Completed')
+    govukTaskListStatusShouldBe(sections.STAFF, 'Completed')
+    govukTaskListStatusShouldBe(sections.WHY_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.EVIDENCE, 'Completed')
+    govukTaskListStatusShouldBe(sections.USUAL_BEHAVIOUR, 'Completed')
+    cy.findAllByText('Incomplete').should('have.length', 2)
+
+    completeTextboxSection(sections.TRIGGERS, /triggers$/, 'What are Testname User’s triggers?')
+
+    cy.findAllByText('Cannot save yet').should('have.length', 1)
+    govukTaskListStatusShouldBe(sections.DETAILS, 'Completed')
+    govukTaskListStatusShouldBe(sections.STAFF, 'Completed')
+    govukTaskListStatusShouldBe(sections.WHY_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.EVIDENCE, 'Completed')
+    govukTaskListStatusShouldBe(sections.USUAL_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.TRIGGERS, 'Completed')
+    cy.findAllByText('Incomplete').should('have.length', 1)
+
+    completeTextboxSection(
+      sections.PROTECTIVE_FACTORS,
+      /protective-factors$/,
+      'What are the protective factors for Testname User?',
+    )
+
+    govukTaskListStatusShouldBe(sections.CHECK, 'Incomplete')
+    govukTaskListStatusShouldBe(sections.DETAILS, 'Completed')
+    govukTaskListStatusShouldBe(sections.STAFF, 'Completed')
+    govukTaskListStatusShouldBe(sections.WHY_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.EVIDENCE, 'Completed')
+    govukTaskListStatusShouldBe(sections.USUAL_BEHAVIOUR, 'Completed')
+    govukTaskListStatusShouldBe(sections.TRIGGERS, 'Completed')
+    govukTaskListStatusShouldBe(sections.PROTECTIVE_FACTORS, 'Completed')
+
+    verifyCya()
+
+    cy.findByText('Confirm and save report').click()
+
+    cy.url().should('to.match', /confirmation$/)
+
+    cy.findByRole('link', { name: 'View all CSIPs for Leeds (HMP)' })
+      .should('be.visible')
+      .and('have.attr', 'href')
+      .and('match', /\/manage-csips\?clear=true$/)
+
+    cy.go('back')
+    cy.url().should('to.match', /confirmation$/)
```

Comment:
as James explained in his PR, it's better to wait a sec before verifying that the redirection logic works.

Reply:
No need to wait, looks like cypress already waits for a redirection to finish before continuing